cli/src/core/parse/profile.go

```diff
@@ -0,0 +1,46 @@
+package parse
+
+import (
+	"cli/core"
+	"cli/core/state"
+
+	"github.com/luno/jettison/errors"
+	"github.com/spf13/cobra"
+)
+
+func GetPackageSpecFromProfile(cmd *cobra.Command, config core.Config, packageSpec core.PackageSpec) (*core.PackageSpec, error) {
+	profile := core.Profile{}
+	profileName, err := cmd.Flags().GetString("profile")
+	if err != nil {
+		return nil, errors.Wrap(err, "")
+	}
+
+	for _, p := range config.Profiles {
+		if p.Name == profileName {
+			profile = p
+			break
+		}
+	}
+
+	if !cmd.Flags().Changed("dev") && profile.Dev {
+		packageSpec.IsDev = profile.Dev
+	}
+
+	if !cmd.Flags().Changed("only") && profile.Only {
+		packageSpec.IsOnly = profile.Only
+	}
+
+	if len(profile.Packages) > 0 {
+		packageSpec.Packages = append(profile.Packages, packageSpec.Packages...)
+	}
+	if len(profile.EnvFiles) > 0 {
+		envViper, err := state.GetEnvironmentVariableViper(profile.EnvFiles)
+		if err != nil {
+			return nil, err
+		}
+		envVariables := state.GetEnvVariableString(envViper)
+		packageSpec.EnvironmentVariables = append(envVariables, packageSpec.EnvironmentVariables...)
```

Comment:
A Q, is that mean that the env vars of the profile will override the other one ?

Reply:
Good question! So, it turns out that this was a potential bug. It was possible that we could end up with more than one instance of the same environment variable. This function's intent isn't to fix that though, so I've added the fix in `cli-v2-pre-prod-6`.